pymodbus/repl/client/main.py

```diff
@@ -111,6 +113,7 @@ def convert(self, value, param, ctx):
 def cli(client): #NOSONAR pylint: disable=too-many-statements,too-many-branches
     """Client definition."""
     use_keys = KeyBindings()
```

Comment:
to me this is overkill, but I do not have a strong opinion (your suggestion work).

Moving it is a good idea, but then why not also make it a command line option as you suggested.

Reply:
That's certainly a better long term solution, but it also makes sense for this to be the default.  You will still want to do the normpath and expand user with the path received from command line option, just to make sure that you are handling the path correctly for the platform.

I'm not familiar with the click library or how you are handling command line options, so it'll be easier for someone else to implement that.

Reply:
+1 on having the ability to change the path of the history file from command line and at the same time we should also be debating if this change needs to be persisted.

Reply:
Command line options really should always fall back to (or equivalently override) a default, rather than persisting between calls - you don't know how long it's going to be between calls, and whether the user will remember what they set it to last (I use detached tmux sessions a lot - sometimes it's weeks between me looking at a particular terminal session).  

But having an environment variable or config file that sets the default would be a good long term aim.